src/libawkward/array/NumpyArray.cpp

```diff
@@ -574,7 +574,24 @@ namespace awkward {
   }
 
   const std::shared_ptr<Content> NumpyArray::flatten(int64_t axis) const {
-    throw std::runtime_error("FIXME: not implemented");
+    std::cout << "My length is " << length() << "\n";
+    std::cout << "My dimentions are " << ndim() << "\n";
+    std::cout << "My num fields are " << numfields() << "\n";
+    std::cout << "My format is " << format() << "\n";
+    std::cout << "My shape size is " << shape().size() << "\n";
+    std::cout << "My strides size is " << strides().size() << "\n";
+    std::cout << "My item size is " << int64_t(itemsize()) << "\n";
+    for(auto& ishape : shape())
+      std::cout << int64_t(ishape) << "\n";
+    for(auto& istride : strides())
+      std::cout << int64_t(istride) << "\n";
+
+    for(int64_t i = 0; i < length(); ++i) {
+      std::shared_ptr<Content> content = getitem_at_nowrap(itemsize());
+      std::cout << " " << i << ": " << content.get()->length() << "\n";
+    }
+
+    throw std::invalid_argument("array contains no Records");
```

Comment:
Here's an example of reshaping a 2×3×5 NumPy array (in Python notation):

```python
>>> a = numpy.arange(2*3*5).reshape(2, 3, 5)
>>> a
array([[[ 0,  1,  2,  3,  4],
        [ 5,  6,  7,  8,  9],
        [10, 11, 12, 13, 14]],

       [[15, 16, 17, 18, 19],
        [20, 21, 22, 23, 24],
        [25, 26, 27, 28, 29]]])
>>> a.reshape((a.shape[0] * a.shape[1],) + a.shape[2:])
array([[ 0,  1,  2,  3,  4],
       [ 5,  6,  7,  8,  9],
       [10, 11, 12, 13, 14],
       [15, 16, 17, 18, 19],
       [20, 21, 22, 23, 24],
       [25, 26, 27, 28, 29]])
```

For `axis=0` (all we're worrying about right now), the first two dimensions get multiplied (`a.shape[0] * a.shape[1]`) and the rest get concatenated (`+ a.shape[2:]`).